pkg/config/parameters.go

```diff
@@ -93,6 +93,22 @@ const IPv4ClusterDNSFamily ClusterDNSFamilyType = "v4"
 const IPv6ClusterDNSFamily ClusterDNSFamilyType = "v6"
 const AllClusterDNSFamily ClusterDNSFamilyType = "all"
 
+// ServerHeaderTransformation defines the action to be applied to the Server header on the response path
+type ServerHeaderTransformationType int32
+
+func (s ServerHeaderTransformationType) Validate() error {
+	switch s {
+	case OVERWRITE, APPEND_IF_ABSENT, PASS_THROUGH:
+		return nil
+	default:
+		return fmt.Errorf("invalid server header transformation %q", s)
+	}
+}
+
+const OVERWRITE ServerHeaderTransformationType = 0
```

Comment:
Again, please make similar changes for the other constants:
```suggestion
const OverwriteServerHeader ServerHeaderTransformationType = "overwrite"
```